src/wallet/wallet.cpp

```diff
@@ -3954,6 +4019,25 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
 
     RegisterValidationInterface(walletInstance);
 
+    // Make sure we have enough keys in our keypool if HD is enabled
+    if (walletInstance->IsHDEnabled()) {
+        unsigned int keypool_size = GetArg("-keypool", DEFAULT_KEYPOOL_SIZE);
+        unsigned int keypool_min = GetArg("-keypoolmin", DEFAULT_KEYPOOL_MIN);
+        if (walletInstance->IsCrypted()) {
+            if (keypool_size < keypool_min) {
+                LogPrintf("Parameter Interaction: keypool size (%d) must be larger than keypool minimum size for encrypted wallets (%d)\n", keypool_size, keypool_min);
+                ForceSetArg("-keypool", std::to_string(keypool_min));
+            }
+        } else {
+            if (keypool_size < keypool_min) {
+                InitWarning(strprintf(_("Your keypool is configured to store %d keys, which is below the keypool minimum size of %d. Using a larger keypool will make it less "
+                                        "likely that your wallet will be missing transactions and funds if it is restored from an old backup."), keypool_size, keypool_min));
+            }
+            walletInstance->TopUpKeyPool();
+        }
+        walletInstance->CheckKeypoolMinSize();
```

Comment:
sorry - I don't understand this comment. Can you explain what you mean by the new error message not quite working?